### PR TITLE
x64: enable VTune support by default

### PR DESCRIPTION
**Pull request by abrown**
After significant work in the `ittapi-rs` crate, this dependency should
build without issue on Wasmtime's supported operating systems: Windows,
Linux, and macOS. The difference in the release binary is <20KB, so this
change makes `vtune` a default build feature. This change upgrades
`ittapi-rs` to v0.2.0 and updates the documentation.

### CODE REVIEW

**Comment by abrown** [comment]
Some more details on this change:
 - I wouldn't mind a second look at what was necessary to implement the multi-OS support in `ittapi-rs`; e.g., right now there are bindings built for each supported OS but unsupported OS-es will cause a build failure because they export nothing (see the [`lib.rs`](https://github.com/intel/ittapi/blob/master/ittapi-rs/src/lib.rs) file)
 - I ran `cargo build --release` before and after this change on a Fedora machine: without the `vtune` feature, the binary was 18,674,952 bytes; with `vtune` on by default, the binary was 18,691,768 bytes. That seemed like an appropriately-small increase.
 - Thanks to @bnjbvr and @jlb6740 for working on this as well!

**Review by bnjbvr — CHANGES_REQUESTED** [review]
Thanks! Having written similar patches to test support on Windows before, this is not sufficient to enable vtune support more widely: there is also a `cfg[target_os = "linux"]` that need to be tweaked in `profiling.rs` iirc. There we should precisely enable this if the target os is linux or windows or mac, to not break compiling to other OS:es.

Then for the default feature: wonder if, in the Cargo.toml file, we can remove it from the list of dependencies (even optional), and then have per-os specific dependencies that both add the dependency and enable the feature?

**Comment by alexcrichton** [comment]
In addition to what @bnjbvr mentioned to make this default I think [this block](https://github.com/bytecodealliance/wasmtime/blob/9c3d5c7b9f446d840e7795a3eba85d95894864f2/crates/wasmtime/Cargo.toml#L53-L63) and [this block](https://github.com/bytecodealliance/wasmtime/blob/9c3d5c7b9f446d840e7795a3eba85d95894864f2/Cargo.toml#L94-L101) need updating to include `vtune`

**Comment by abrown** [comment]
> cfg[target_os = "linux"] that need to be tweaked in profiling.rs iirc

This is resolved by the latest commit.

> we can remove it from the list of dependencies (even optional), and then have per-os specific dependencies that both add the dependency and enable the feature

We could, yes, but it seems like we should do one or the other: either enable the `vtune` feature exclusively in Cargo like you suggest OR have the `cfg(...)` conditions in the code itself. This could be resolved in a later PR, I think. 

> I think [this block](https://github.com/bytecodealliance/wasmtime/blob/9c3d5c7b9f446d840e7795a3eba85d95894864f2/crates/wasmtime/Cargo.toml#L53-L63) and [this block](https://github.com/bytecodealliance/wasmtime/blob/9c3d5c7b9f446d840e7795a3eba85d95894864f2/Cargo.toml#L94-L101) need updating to include vtune

One of those was already "defaulted" by this PR but I added a default to `wasmtime-jit` to be consistent.

**Comment by abrown** [comment]
Oh, hold on: we also need to check for `x86_64` in `cfg(...)` as well, right?

**Comment by alexcrichton** [comment]
Oh reading more closely ... personally I think it's ok to skip platform-specific checks in the #[cfg] because Wasmtime probably only really compiles on Linux/macOS/Windows anyway and other platforms likely hit compile errors elsewhere so this wouldn't be the only issue. If Wasmtime is ported to other platforms I think it'd be good to leave this in as "hey this crate probably needs an update" if possible

**Comment by abrown** [comment]
@bnjbvr, @alexcrichton: this removes the conditional compilation for the OS based on @alexcrichton's comment--this way, if someone ports Wasmtime to another OS but does not turn the `vtune` feature off, they will get a compile error in `ittapi-rs` which will motivate adding the OS support there.

I did add a `target_arch = "x86_64"` check since this feature only makes sense to be enabled on those machines anyways. `ittapi-rs` seems to compile fine on other architectures so there should be no problem there but this makes it more clear what environment the `vtune` feature is intended for.

File: crates/jit/Cargo.toml [hunk 1]
```diff
@@ -35,6 +35,7 @@ winapi = { version = "0.3.8", features = ["winnt", "impl-default"] }
 rustix = "0.33.0"
 
 [features]
+default = ['vtune']
```
[the remark this answers is not in the document]
Comment (abrown) [reply crates/jit/Cargo.toml:38]: Ok, removing...

File: docs/examples-profiling-vtune.md [hunk 2]
```diff
@@ -30,13 +30,10 @@ For more information on VTune and the analysis tools it provides see its
...
-
-```sh
-$ cargo build --features=vtune
-```
-
-Then, enable runtime support based on how you use Wasmtime:
+For JIT profiling with VTune, Wasmtime currently builds with the `vtune` feature
+enabled by default. This ensures the compiled binary understands how to inform
+the `ittapi` library of JIT events. But it must still be enabled at
+runtime--enable runtime support based on how you use Wasmtime:
```
Comment (bnjbvr) [inline docs/examples-profiling-vtune.md:36]: hmm, it seems the end of this comment after `--` would need to be removed, as it contains `:` but with no extra indication thereafter?

**Review by bnjbvr — APPROVED** [review]
Works fine with the wasmtime CLI binary on Linux! Thanks.

File: docs/examples-profiling-vtune.md (hunk 2 above)
Comment (abrown) [reply docs/examples-profiling-vtune.md:36]: I'll leave it and we can fix it up later if we think of a more clear way to word this.

Comment (bnjbvr) [reply docs/examples-profiling-vtune.md:36]: Oh, i misread it as a list entry as well, my bad. Maybe can reformulate slightly then:

```suggestion
the `ittapi` library of JIT events. But it must still be enabled at
runtime with one of the following:
```

**Pull request merged**